packages/go_router/lib/src/matching.dart

```diff
@@ -134,14 +140,99 @@ class RouteMatchList {
   bool get isError => matches.length == 1 && matches.first.error != null;
 
   /// Returns the error that this match intends to display.
-  Exception? get error => matches.first.error;
+  Exception? get error => matches.firstOrNull?.error;
 
   @override
   String toString() {
     return '${objectRuntimeType(this, 'RouteMatchList')}($fullpath)';
   }
 }
 
+/// Unmodifiable version of [RouteMatchList].
+@immutable
+class UnmodifiableRouteMatchList implements RouteMatchList {
```

Comment:
This doesn't seem necessary as the _routeMatchList is not expose externally, no? I will just use a RouteMatchList

Reply:
This is needed since we need to store a copy of a RouteMatchList for each branch, to be able to restore it when calling goBranch. Since RouteMatchList is mutable, this copy also needs to contain copies for fields like _matches etc. Not sure if an unmodifiable version is strictly speaking still necessary though, will check if a copy of RouteMatchList will suffice.